### PR TITLE
fix: don't use custom error document for 403

### PR DESCRIPTION
**Pull request by ottokruse**
fix: don't use custom error document for 403 as this exposes index.html on POST requests

*Issue #, if available:*

#25 

*Description of changes:*

- Use DefaultRootObject
- Add s3:ListBucket to bucket policy
- Return 403 from Lambda (this is safe now)

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: src/lambda-edge/parse-auth/index.ts [hunk 1]
```diff
@@ -50,7 +50,7 @@ export const handler: CloudFrontRequestHandler = async (event) => {
     } catch (err) {
         return {
             body: createErrorHtml('Bad Request', err.toString(), redirectedFromUri),
-            status: '400', // Note: do not send 403 (!) as we have CloudFront send back index.html for 403's to enable SPA-routing
+            status: '400',
```
Comment (basile-henry) [inline src/lambda-edge/parse-auth/index.ts]: Should this be `403` now?

Comment (ottokruse) [reply src/lambda-edge/parse-auth/index.ts]: It doesn't matter for our purpose.

Maybe a matter of taste but I think 400 is clear as the request is indeed "bad".

The regular 403 redirects to Cognito. 400 then means here: we're not even gonna redirect you, your request is messed up.

Comment (ottokruse) [reply src/lambda-edge/parse-auth/index.ts]: I deployed the new version using SAM and tested it, all good. Also deployed through serverless application repository, all good.

I'm gonna merge this PR now. THANK YOU for reporting the issue.

Comment (basile-henry) [reply src/lambda-edge/parse-auth/index.ts]: Thank you for your quick fixes! :+1: :smile:

**Pull request merged**